src/auth/KeychainStorage.ts

```diff
@@ -0,0 +1,59 @@
+import {Platform} from 'react-native'
+import * as Keychain from 'react-native-keychain'
+
+async function write(key: string, value: string) {
+  return Keychain.setGenericPassword(key, value, {
+    service: key,
+    accessControl: Keychain.ACCESS_CONTROL.BIOMETRY_ANY_OR_DEVICE_PASSCODE,
+    accessible: Keychain.ACCESSIBLE.WHEN_UNLOCKED_THIS_DEVICE_ONLY,
+    securityLevel: Keychain.SECURITY_LEVEL.SECURE_HARDWARE,
+  }).then((result) => {
+    if (result === false) Promise.reject('Unable to store secret')
```

Comment:
reject with error, not a string?

Reply:
whatever is thrown here will be translated into an unknown error.